pkg/operationreport/externalerror.go

```diff
@@ -293,3 +293,27 @@ func ErrTypeBodyMustNotBeEmpty(definitionType, typeName string) (err ExternalErr
 	err.Message = fmt.Sprintf("the %s named '%s' is invalid due to an empty body", definitionType, typeName)
 	return err
 }
+
+func ErrKeyDirectiveMustHaveSingleArgument(typeName string) (err ExternalError) {
+	err.Message = fmt.Sprintf("the key directive on the entity named '%s' must have a single argument named 'fields'", typeName)
+	return err
+}
+
+func ErrPrimaryKeyReferencesMustExistOnEntity(primaryKey, typeName string) (err ExternalError) {
```

Comment:
I will vote for 2 separate methods for errors

```
ErrEntityPrimaryKeyIsEmpty
ErrEntityPrimaryKeyReferencesMustExist
```

Reply:
OK